lib/chef/provider/aws_security_group.rb

```diff
@@ -1,83 +1,288 @@
 require 'chef/provisioning/aws_driver/aws_provider'
 require 'date'
 require 'ipaddr'
+require 'set'
 
 class Chef::Provider::AwsSecurityGroup < Chef::Provisioning::AWSDriver::AWSProvider
 
   action :create do
     sg = new_resource.aws_object
-    if !sg
-      converge_by "Creating new SG #{new_resource.name} in #{region}" do
-        options = { description: new_resource.description }
-        options[:vpc] = new_resource.vpc if new_resource.vpc
-        options = AWSResource.lookup_options(options, resource: new_resource)
-        Chef::Log.debug("VPC: #{options[:vpc]}")
-
-        sg = driver.ec2.security_groups.create(new_resource.name, options)
-      end
+    if sg
+      update_security_group(sg)
+    else
+      sg = create_security_group
     end
 
     new_resource.save_managed_entry(sg, action_handler)
```

Comment:
In the [aws_vpc](https://github.com/chef/chef-provisioning-aws/blob/27f4b4b9c8532b80b87996e87861b77202bf4551/lib/chef/provider/aws_vpc.rb#L74) provider you're only saving the managed entry if we fall into the create block.  That prevents creating a managed entry if the user references the VPC by `vpc_id`.  Shouldn't we have the same behavior here?